frontend/public/components/hpa.tsx

```diff
@@ -31,8 +34,113 @@ const menuActions = [
+  if (external.targetAverageValue) {
+    currentValue = _.get(current, 'object.currentAverageValue');
+    targetValue = external.targetAverageValue;
+  } else if (external.targetValue) {
+    currentValue = _.get(current, 'object.currentValue');
+    targetValue = external.targetValue;
+  }
+
+  return <MetricsRow key={key} type={type} current={currentValue} target={targetValue} />;
+};
+
+const objectRow = (metric, current, key, ns) => {
+  const { object } = metric;
+  const type = <React.Fragment>
+    {object.metricName} on
+    <ResourceLink kind={object.target.kind} name={object.target.name} namespace={ns} title={object.target.name} />
+  </React.Fragment>;
+  const currentValue = _.get(current, 'object.currentValue');
+  const targetValue = object.targetValue;
+
+  return <MetricsRow key={key} type={type} current={currentValue} target={targetValue} />;
+};
+
+const podRow = (metric, current, key) => {
+  const { pods } = metric;
+  const type = `${pods.metricName} on pods`;
+  const currentValue = _.get(current, 'pods.currentAverageValue');
+  const targetValue = pods.targetAverageValue;
+
+  return <MetricsRow key={key} type={type} current={currentValue} target={targetValue} />;
+};
+
+const resourceRow = (metric, current, key) => {
+  let currentValue, targetValue;
+  let type = <React.Fragment>resource {metric.resource.name}</React.Fragment>;
+  if (metric.resource.targetAverageUtilization) {
+    type = <React.Fragment>
+      {type}&nbsp;<span className="small text-muted">(as a percentage of request)</span>
+    </React.Fragment>;
+    const utilization = _.get(current, 'resource.currentAverageUtilization');
+    if (_.isNumber(utilization)) {
+      currentValue = `${utilization}%`;
+    }
+    targetValue = `${metric.resource.targetAverageUtilization}%`;
+  } else if (metric.resource.targetAverageValue) {
+    currentValue = _.get(current, 'resource.currentAverageValue');
+    targetValue = metric.resource.targetAverageValue;
+  }
+
+  return <MetricsRow key={key} type={type} current={currentValue} target={targetValue} />;
+};
+
+const MetricsTable: React.SFC<MetricsTableProps> = ({obj: hpa}) => {
+  return <React.Fragment>
+    <h1 className="co-section-title">Metrics</h1>
+    <div className="co-m-table-grid co-m-table-grid--bordered">
+      <div className="row co-m-table-grid__head">
+        <div className="col-xs-6">Type</div>
+        <div className="col-xs-3">Current</div>
+        <div className="col-xs-3">Target</div>
+      </div>
+      <div className="co-m-table-grid__body">
+        {hpa.spec.metrics.map((metric, i) => {
+          // https://github.com/kubernetes/api/blob/master/autoscaling/v2beta1/types.go
+          const current = _.get(hpa, ['status', 'currentMetrics', i]);
+          switch (metric.type) {
+            case 'External':
+              return externalRow(metric, current, i);
+            case 'Object':
+              return objectRow(metric, current, i, hpa.metadata.namespace);
```

Comment:
Any reason `key` is not the last arg for just this one?

Reply:
Passing `key` to `fn`s seems fine, I think.  Had a short chat w/@spadgett about it.

Reply:
Yeah I think putting key last always probably makes sense